jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java

```diff
@@ -174,36 +190,80 @@
+                        .collect(Collectors.toList());
+                if (referenceLibraries.size() > 0) {
+                    result.add(new ContainerNode(REFERENCED_LIBRARIES_CONTAINER_NAME, REFERENCED_LIBRARIES_PATH, NodeKind.CONTAINER,
+                            IClasspathEntry.CPE_CONTAINER));
+                }
+                return result;
             } catch (CoreException e) {
                 JdtlsExtActivator.logException("Problem load project library ", e);
             }
         }
         return Collections.emptyList();
     }
 
+    /**
+     * Get the correspond node of classpath, it may be container or a package root
+     *
+     * @param classpathEntry
+     *            classpath
+     * @param javaProject
+     *            correspond java project
+     * @param nodeKind
+     *            could be CONTAINER or PACKAGEROOT(for referenced libraries)
+     * @return
+     */
+    private static PackageNode getNodeFromClasspathEntry(IClasspathEntry classpathEntry, IJavaProject javaProject, NodeKind nodeKind) {
+        try {
+            IClasspathEntry entry = JavaCore.getResolvedClasspathEntry(classpathEntry);
+            IClasspathContainer container = JavaCore.getClasspathContainer(entry.getPath(), javaProject);
+            // HACK: There is an initialization issue for the first container.
+            if (container == null) {
+                container = JavaCore.getClasspathContainer(entry.getPath(), javaProject);
+            }
+            if (container != null) {
+                switch (nodeKind) {
+                case CONTAINER:
+                    return new ContainerNode(container.getDescription(), container.getPath().toPortableString(), nodeKind, entry.getEntryKind());
+                case PACKAGEROOT:
+                    // Use package name as package root name
+                    String[] pathSegments = container.getPath().segments();
+                    return new PackageRootNode(pathSegments[pathSegments.length - 1], container.getPath().toPortableString(), nodeKind,
+                            IClasspathEntry.CPE_PROJECT);
+                default:
+                    return null;
+                }
+            }
+        } catch (CoreException e) {
+            e.printStackTrace();
+            // Ignore it
+        }
+        return null;
+    }
+
+    private static PackageNode getNodeFromClasspathVariable(IClasspathEntry classpathEntry) {
+        IClasspathEntry entry = JavaCore.getResolvedClasspathEntry(classpathEntry);
+        String name = classpathEntry.getPath().toPortableString();
+        String path = entry.getPath().toPortableString();
+        if (new File(path).isDirectory()) {
+            return new PackageRootNode(name, path, NodeKind.PACKAGEROOT, IClasspathEntry.CPE_LIBRARY);
+        } else {
+            return new PackageRootNode(name, path, NodeKind.PACKAGEROOT, IClasspathEntry.CPE_PROJECT);
+        }
```

Comment:
why IClasspathEntry.CPE_PROJECT ?

Reply:
Follow by the behavior of Maven references, jar was set to IClasspathEntry.CPE_PROJECT